hack/tools/release/notes.go

```diff
@@ -223,53 +225,49 @@ func run() int {
 		}
 	}
 
-	for _, c := range commits {
-		body := trimTitle(c.body)
-		var key, prNumber, fork string
-		prefix, err := getAreaLabel(c.merge)
-		if err != nil {
-			fmt.Println(err)
-			os.Exit(1)
-		}
-		switch {
-		case strings.HasPrefix(body, ":sparkles:"), strings.HasPrefix(body, "✨"):
-			key = features
-			body = strings.TrimPrefix(body, ":sparkles:")
-			body = strings.TrimPrefix(body, "✨")
-		case strings.HasPrefix(body, ":bug:"), strings.HasPrefix(body, "🐛"):
-			key = bugs
-			body = strings.TrimPrefix(body, ":bug:")
-			body = strings.TrimPrefix(body, "🐛")
-		case strings.HasPrefix(body, ":book:"), strings.HasPrefix(body, "📖"):
-			key = documentation
-			body = strings.TrimPrefix(body, ":book:")
-			body = strings.TrimPrefix(body, "📖")
-			if strings.Contains(body, "CAEP") || strings.Contains(body, "proposal") {
-				key = proposals
+	results := make(chan processCommitResult)
+	jobs := make(chan *commit)
+	var wg sync.WaitGroup
+
+	wg.Add(*numWorkers)
+	for i := 0; i < *numWorkers; i++ {
+		go func() {
+			for commit := range jobs {
+				processed := processCommitResult{}
+				processed.prEntry, processed.err = generateNotesPREntry(commit)
+				results <- processed
```

Comment:
```suggestion
				results <- generateNotesPREntry(commit)
```
if we change `generateNotesPREntry` to return a `processCommitResult`

Reply:
I would prefer to keep this
It makes the function agnostic from the fact that it's being called from concurrent routines. And delegates the aggregation of result an error to the routine itself.

I can add another method to do this encapsulation, but that seemed like extra overhead for saving two lines.

Reply:
ack, fair